src/lib/storage/table.cpp

```diff
@@ -200,7 +205,10 @@ void Table::append_chunk(const Segments& segments, std::shared_ptr<MvccData> mvc
   }
 #endif
 
-  _chunks.push_back(std::make_shared<Chunk>(segments, mvcc_data, alloc));
+  auto c = std::make_shared<Chunk>(segments, mvcc_data, alloc);
+  _chunks.push_back(c);
+
+  if (mvcc_data) c->update_max_begin_cid();
```

Comment:
This does not really feel like something that the table should do

Reply:
I could imagine a periodic task that does that. However, this is not going to happen soon. Just remove it here?

Reply:
How about a `finalize_chunk` (name tbd) method that
* sets Chunk::_is_mutable (that should not be the job of the encoder)
* sets the max_begin_cid
* potentially encodes the chunk or schedules it for encoding
* potentially builds indexes, converts indexes, or schedules a job doing that
* ...